app/policies/partner_policy.rb

```diff
@@ -60,11 +60,14 @@ class Scope < Scope
     def resolve
       if user.root?
         scope.all
+
       else
-        scope.left_outer_joins(:users, :address)
-             .where('partners_users.user_id = ? OR addresses.neighbourhood_id IN (?)',
-                    user.id, user.owned_neighbourhood_ids)
-             .distinct
+        user_neighbourhood_ids = user.owned_neighbourhood_ids
+
+        scope.left_outer_joins(:users, :address, :service_areas)
+          .where('partners_users.user_id = ? OR addresses.neighbourhood_id IN (?) OR service_areas.neighbourhood_id IN (?)',
```

Comment:
maybe this query can be split into multiple lines?

Reply:
not sure `where` is good at handling `or`-ed terms. especially with left joins

Reply:
assume aster means to just break up the string for readability? :)